src/App/Datums.cpp

```diff
@@ -382,7 +367,8 @@ bool LocalCoordinateSystem::LCSExtension::extensionGetSubObject(DocumentObject*&
     }
 
     try {
-        ret = obj->getDatumElement(name.c_str());
+        auto* lcs = dynamic_cast<const LocalCoordinateSystem*>(getExtendedObject());
```

Comment:
Please use `isDerivedFrom`, see explanation here https://github.com/FreeCAD/FreeCAD/pull/20476#discussion_r2018462898

Reply:
If the cast can fail, calling upon a nullptr will cause segfault, and nullptr check is needed. If the cast can never fail, it would be static_cast

Reply:
Here we don't need the lcs variable anymore since the extension is now correctly set to the LCS object